src/main/java/seedu/address/storage/JsonAddressBookStorage.java

```diff
@@ -56,6 +57,9 @@ public Optional<ReadOnlyAddressBook> readAddressBook(Path filePath) throws DataL
         } catch (IllegalValueException ive) {
             logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
             throw new DataLoadingException(ive);
+        } catch (CompanyNotFoundException cnfe) {
+            logger.info("Job associated with non-existent company in " + filePath + ": " + cnfe.getMessage());
+            throw new DataLoadingException(cnfe);
```

Comment:
Nice. I will probably fix the non-existing match issue that KH raised for me in the same way as this.